File: sesac7_flutter_recipe_app/lib/data/mapper/ingredients_mapper.dart
import 'package:flutter_recipe_app/data/dto/ingredients_dto.dart';
import 'package:flutter_recipe_app/data/mapper/ingredient_mapper.dart';
import 'package:flutter_recipe_app/domain/model/ingredients.dart';

extension IngredientsDtoToModel on IngredientsDto {
  Ingredients toModel() {
    return Ingredients(
      ingredient: ingredient!.toModel(),
      amount: amount!.toInt(),
    );
  }
}

extension IngredientsModelToDto on Ingredients {
  IngredientsDto toDto() {
    return IngredientsDto(
      ingredient: ingredient.toDto(),
      amount: amount.toInt(),
    );
  }
}

File: compose_recipe/app/src/main/java/com/example/recipeapp/data/mapper/IngredientsMapper.kt
package com.example.recipeapp.data.mapper

import com.example.recipeapp.data.dto.IngredientsDto
import com.example.recipeapp.data.model.Ingredients

fun IngredientsDto.toModel(): Ingredients {
    return Ingredients(
        ingredient = this.ingredient!!.toModel(),
        amount = this.amount!!
    )
}

fun Ingredients.toDto(): IngredientsDto {
    return IngredientsDto(
        ingredient = this.ingredient.toDto(),
        amount = this.amount
    )
}